eth/tracers/api.go

```diff
@@ -799,10 +799,19 @@ func containsTx(block *types.Block, hash common.Hash) bool {
 // TraceTransaction returns the structured logs created during the execution of EVM
 // and returns them as a JSON object.
 func (api *API) TraceTransaction(ctx context.Context, hash common.Hash, config *TraceConfig) (interface{}, error) {
-	_, blockHash, blockNumber, index, err := api.backend.GetTransaction(ctx, hash)
+	// GetTransaction returns 0 for the blocknumber if the transaction is not found
+	tx, blockHash, blockNumber, index, err := api.backend.GetTransaction(ctx, hash)
 	if err != nil {
 		return nil, err
 	}
+	if tx == nil {
+		var histResult []*txTraceResult
```

Comment:
As discussed today, we may not be able to directly forward this result because `TraceTransaction` is expect to return a single json marshalled logger.ExecutionResult while there appears to be a wrapping layer on top of this function which returns an array of execution results.

Reply:
So, to be clear you think that these changes to `TraceTransaction` will require a similar treatment to the changes in `TraceCall` (#21)?

Reply:
maybe, I think I got confused here

Reply:
After looking at this a bit more closely, I'm surprised that this type is different than the return type in `TraceCall`